core/rfb.js

```diff
@@ -1069,53 +1172,141 @@
+                Util.Warn("Intel AMT KVM only supports 8/16 bit depths, using server pixel format");
+                this._convert_color = true;
+            }
+
+            // ATEN 'wisdom' from chicken-aten-ikvm:lens/lens.rb
+            // tested against the following Supermicro motherboards
+            // (use 'dmidecode -s baseboard-product-name' for model):
+            // - X7SPA-F
+            // - X8DTL
+            // - X8SIE-F
+            // - X9SCL/X9SCM
+            // - X9SCM-F
+            // - X9DRD-iF
+            // - X9SRE/X9SRE-3F/X9SRi/X9SRi-3F
+            // - X9DRL-3F/X9DRL-6F
+            // - X10SLD
+            //
+            // Supported using the ATEN "AST2100" encoding (0x57 / 87):
+            // - X10SL7-F
+            // - X10SLD-F
+            // - X10SLM-F
+            // - X10SLE
+            //
+            // Simply does not work:
+            // Hermon (WPMC450) [hangs at login]:
+            // - X7SB3-F
+            // - X8DTU-F
+            // - X8STi-3F
+            // Peppercon (Raritan/Kira100) [connection refused]:
+            // - X7SBi
+            //
+            // Thanks to Brian Rak and Erik Smit for testing
+            if (this._rfb_atenikvm) {
+                // we do not know the resolution till the first fbupdate so go large
+                // although, not necessary, saves a pointless full screen refresh
+                this._fb_width                = RFB.ATEN_INIT_WIDTH;
+                this._fb_height               = RFB.ATEN_INIT_HEIGHT;
+
+                // TODO(kelleyk): This message (and this block of code) is part
+                // of the original ATEN "HERMON" (0x59) support.  The "AST2100"
+                // (0x57) encoding delivers RGB888 color.  I suppose that we should
+                // update this somehow?  What effect does it have?
+                Util.Warn("ATEN iKVM lies and only does 15 bit depth with RGB555");
+                this._convert_color            = true;
+                this._pixelFormat.bpp         = 16;
+                this._pixelFormat.depth       = 15;
+                this._pixelFormat.red_max     = (1 << 5) - 1;
+                this._pixelFormat.green_max   = (1 << 5) - 1;
+                this._pixelFormat.blue_max    = (1 << 5) - 1;
+                this._pixelFormat.red_shift   = 10;
+                this._pixelFormat.green_shift = 5;
+                this._pixelFormat.blue_shift  = 0;
+
+                // XXX(kelleyk): Doing this will break interaction with non-ATEN
+                // servers until the page is reloaded; it also breaks the mouse/
+                // keyboard portions of the test suite!
+                // 
+                RFB.messages.keyEvent = RFB.messages.atenKeyEvent;
+                RFB.messages.pointerEvent = RFB.messages.atenPointerEvent;
+            }
+
+            if (this._convert_color)
+                this._display.set_true_color(this._pixelFormat.true_color);
```

Comment:
Won't the display always be true color after these changes? If so, just nuke everything regarding color maps from the display code.